lib/semian/sysv_state.rb

```diff
@@ -0,0 +1,59 @@
+require 'forwardable'
+
+module Semian
+  module SysV
+    class State < Semian::Simple::State #:nodoc:
+      include SysVSharedMemory
+      extend Forwardable
+
+      def_delegators :@integer, :semid, :shmid, :synchronize, :transaction,
+                     :shared?, :acquire_memory_object, :bind_init_fn
+      private :shared?, :acquire_memory_object, :bind_init_fn
+
+      def initialize(name:, permissions:)
+        @integer = Semian::SysV::Integer.new(name: name, permissions: permissions)
+        initialize_lookup([:closed, :open, :half_open])
+      end
+
+      def open
+        self.value = :open
+      end
+
+      def close
+        self.value = :closed
+      end
+
+      def half_open
+        self.value = :half_open
+      end
```

Comment:
I don't want `#value=` to be the public API. I don't give a shit about internal private APIs.